src/pydata_sphinx_theme/assets/scripts/index.js

```diff
@@ -43,23 +43,25 @@ function scrollToActive() {
   );
 
   if (!isNaN(storedScrollTop)) {
+    // If we've got a saved scroll position, just use that
     sidebar.scrollTop = storedScrollTop;
+    console.log("[PST]: Scrolled sidebar using stored browser position...");
   } else {
-    var active_pages = sidebar.querySelectorAll(".active");
-    var offset = 0;
-    var i;
-    for (i = active_pages.length - 1; i > 0; i--) {
-      var active_page = active_pages[i];
-      if (active_page !== undefined) {
-        offset += active_page.offsetTop;
+    // Otherwise, calculate a position to scroll to based on the lowest `active` link
+    var sidebarNav = document.getElementById("bd-docs-nav");
+    var active_pages = sidebarNav.querySelectorAll(".active");
+    if (active_pages.length > 0) {
+      // Use the last active page as the offset since it's the page we're on
+      var latest_active = active_pages[active_pages.length - 1];
+      var offset =
+        latest_active.getBoundingClientRect().y -
+        sidebar.getBoundingClientRect().y;
+      // Only scroll the navbar if the active link is lower than 50% of the page
+      if (offset > sidebar.clientHeight * 0.5) {
+        sidebar.scrollTop = offset - sidebar.clientHeight * 0.3;
```

Comment:
Where the `0.3` is coming from?

Reply:
just a little buffer so that the active page doesn't scroll to the very top, but is slightly centered on the screen

Reply:
In the latest commit I've explicitly defined this as a variable so it's more clear